src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NETCOREAPP2_0
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Build.Locator
+{
+    internal static class DotNetSdkLocationHelper
+    {
+        private static readonly Regex DotNetBasePathRegex = new Regex("Base Path:(.*)$", RegexOptions.Multiline);
+
+        public static VisualStudioInstance GetInstance(string workingDirectory)
+        {
+            string dotNetSdkPath = GetDotNetBasePath(workingDirectory);
+
+            if (string.IsNullOrWhiteSpace(dotNetSdkPath))
+            {
+                return null;
+            }
+
+            if (!File.Exists(Path.Combine(dotNetSdkPath, "Microsoft.Build.dll")))
+            {
+                return null;
+            }
+
+            FileInfo dotNetAssemblyPath = new FileInfo(Path.Combine(dotNetSdkPath, "dotnet.dll"));
```

Comment:
It's a little unfortunate to take the `dotnet.dll` version here, since that's tied to the .net core _runtime_ version, where the interesting thing here is the _sdk_ version.

Reply:
I was hoping that the SDK version would be easy to extract from `dotnet --info` but it's not, really:

```
S:\Locator>dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview-009812
 Commit:    e3abf6e935

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview-009812\
```

I don't want to assume that there would never be anything between the `.NET Core SDK` line and the `Version:` line, and `Version:` isn't unique  (there's a host version too). So I guess this is ok for now.

Reply:
What is this version used for? The version of dotnet.dll is not related to the runtime, it is the version of dotnet/cli repo (for 2.x this is the same as the sdk version, for 3.x it is not.)

Reply:
The SDK version is relatively easy to extract from the .version file in the Base Path.

Reply:
Also, for `dotnet --info` to locate an SDK, the unqualified directory name will be its version.

Reply:
Not sure which implementation detail is better to depend upon. There is also the fact that the API here wants a System.Version and not a semantic version.

Reply:
@rainersigwald So this code is looking at the dotnet.dll within the SDK directory. Opening file properties on a few of these locally shows the FileVersion matches the dotnet sdk version. 
![image](https://user-images.githubusercontent.com/611219/52598629-38902700-2e0b-11e9-8de5-fcbc233269f7.png)

Reply:
Yeah, this is more and more confusing: 
![image](https://user-images.githubusercontent.com/3347530/52600552-962f6e00-2e21-11e9-92f1-c6832e06f1ff.png)

Maybe we could just feed the name of the folder into the VS instance name? I also lamented that we have a `System.Version` rather than something useful in that structure.

Reply:
Right, as I said "for 2.x this is the same as the sdk version, for 3.x it is not". In 3.x dotnet.dll is built in dotnet/cli repo and inserted into dotnet/core-sdk as a binary. The version number of dotnet/cli is not a customer facing number.

Reply:
(Complete aside: that light properties dialog over a dark explorer is bothering my eyes.)

Reply:
@nguerrera Took your suggestion to use the .version file. Thanks